### PR TITLE
feat(nextjs): Next.js@15 compatibility

### PR DESCRIPTION
**Pull request by jacekradko**
## Description

Next.js@15 compatibility integration branch

## Checklist

- [ ] `npm test` runs as expected.
- [ ] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

### CODE REVIEW

**Comment by drosi94** [comment]
Is there anything to be done? We could help to push it on main as the Next.JS just released v15

**Comment by jacekradko** [comment]
!allow-major

**Review by dstaley — APPROVED** [review]

File: .changeset/grumpy-hairs-remember.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+```typescript
+import { auth } from '@clerk/nextjs/server';
+
+export function GET() {
+  const { userId } = auth();
+  return new Response(JSON.stringify({ userId }));
+}
+```
+
+The `auth`method now becomes asynchronous. You will need to make the following changes to the snippet above to make it compatible.
```
Comment (jacekradko) [inline .changeset/grumpy-hairs-remember.md:24]:
```suggestion
The `auth` method now becomes asynchronous. You will need to make the following changes to the snippet above to make it compatible.
```

**Review by BRKalow — APPROVED** [review]
🚢

**Comment by ChristianIvicevic** [comment]
I have tested the canary for this and noticed that the types for the `clerkMiddleware` aren't correct yet.
```ts
export default clerkMiddleware(async (auth, request) => {
  const { sessionClaims } = await auth() // auth() is not a Promise over here and causes TS and eslint to complain
  // ...
  if (!isPublicRoute(request)) {
    try { await auth.protect() } // works as expected
    catch (e) { unstable_rethrow(e) }
  }
})
```
Do you want me to open a separate issue to track this?

**Comment by dstaley** [comment]
@ChristianIvicevic Thanks for reporting this! It was fixed in [`6.0.0-canary.vee453d8`](https://www.npmjs.com/package/@clerk/nextjs/v/6.0.0-canary.vee453d8).

**Pull request merged**